### PR TITLE
fix(typescript-estree): parsing of deeply nested new files in new folder

### LINKED ISSUE #1394: IDE Parsing error when using overlapping globs
**What code were you trying to parse?**
New files added to the project

**What did you expect to happen?**
ESLint works

**What actually happened?**
```
Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: src/ui/components/core/DragHandle/DragHandle.tsx.
The file must be included in at least one of the projects provided.
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.10.0` |
| `TypeScript`                | `3.7.4` |
| `ESLint`                    | `6.7.2` |
| `node`                      | `12.8.0` |
| `npm`                       | `6.13.4` |

`.eslintrc.js`

```javascript
module.exports = {
	plugins: ["import"],
	extends: ["react-app", "plugin:import/errors", "plugin:import/warnings", "plugin:import/typescript", "prettier"],
	parserOptions: {
		project: "./tsconfig.eslint.json",
	},
	rules: {
		"import/default": "off", // not working with TS right now for export = (https://github.com/benmosher/eslint-plugin-import/issues/1527)
		"import/named": "off", // not needed with TS.
		"import/namespace": "off", // not needed with TS.
		"import/no-cycle": "off", // takes long, using madge for this
		"import/no-internal-modules": "off",
		"import/order": [
			"error",
			{
				"newlines-between": "always",
			},
		],
		"no-shadow": ["error"],
		"prefer-const": [
			"warn",
			{
				destructuring: "all",
			},
		],
		"require-await": "off",
	},
	overrides: [
		{
			files: ["**/*.ts?(x)"],
			rules: {
				"@typescript-eslint/ban-ts-ignore": "warn",
				"@typescript-eslint/explicit-function-return-type": "off",
				"@typescript-eslint/interface-name-prefix": [
					"warn",
					{
						prefixWithI: "always",
						allowUnderscorePrefix: false,
					},
				],
				"@typescript-eslint/no-inferrable-types": "off",
				"@typescript-eslint/require-await": "warn",
			},
		},
	],
	settings: {
		"import/extesions": [".js", ".ts", ".tsx"],
		"import/parsers": {
			"@typescript-eslint/parser": [".ts", ".tsx"],
		},
	},
```

`tsconfig.eslint.json`
```json
{
	"extends": "./tsconfig",
	"include": ["./**/*", "./.storybook/**/*"],
	"exclude": ["assets", "build", "build-resources", "node_modules", "tmp"]
}
```

`tsconfig.json`
```json
{
	"compilerOptions": {
		"target": "es6",
		"lib": ["dom", "dom.iterable", "esnext"],
		"allowJs": false,
		"skipLibCheck": true,
		"esModuleInterop": true,
		"allowSyntheticDefaultImports": true,
		"strict": true,
		"forceConsistentCasingInFileNames": true,
		"module": "esnext",
		"moduleResolution": "node",
		"resolveJsonModule": true,
		"isolatedModules": true,
		"noEmit": true,
		"jsx": "preserve",
		"downlevelIteration": true,
		"sourceMap": true
	},
	"include": ["src"],
	"exclude": ["src/electron"]
}
```

`.env` (for create-react-app)
```
EXTEND_ESLINT=true
```
The only fix this to restart the ESLint process in IntelliJ. Once it's restarted, eslint works just fine on all files in `src` no matter how deeply nested. As soon as I create a new file it comes back.
Looks related to https://github.com/typescript-eslint/typescript-eslint/issues/1303#issue-532524423

### PR DESCRIPTION
**Pull request by armano2**
### Context:
Currently `**` is resolved only as directory, it means that for `./**/*` paths `/src/test.ts`, `/test.ts` is going to work but `/src/test/test.ts` will not.

This is edge case and happens when you setup includes with `**`.

### Config example:

```json5
// tsconfig.extend.json
{
  "include": ["src"],
}
```
```json5
// tsconfig.json
{
  "extends": "./tsconfig.extend.json",
  "include": ["./**/*"],
}
```

fixes #1394

### CODE REVIEW

File: packages/typescript-estree/tests/lib/persistentParse.ts [hunk 1]
```diff
@@ -230,4 +240,37 @@ describe('persistent parse', () => {
 
     baseTests(tsConfigExcludeBar, tsConfigIncludeAll);
   });
+
+  describe('ui/components/file', () => {
```
Comment (armano2) [inline packages/typescript-estree/tests/lib/persistentParse.ts]: i know, i know name is bad, i'm just testing

File: packages/typescript-estree/tests/lib/persistentParse.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ const CONTENTS = {
   foo: 'console.log("foo")',
   bar: 'console.log("bar")',
   'baz/bar': 'console.log("baz bar")',
+  'ui/components/file': 'console.log("ui/components")',
```
Comment (bradzacher) [inline packages/typescript-estree/tests/lib/persistentParse.ts]: hmmm.
I think that you could probably achieve this via fixing the "allows parsing of deeply nested new files" test in `baseTests`.

I.e. change `baz/bar` to `baz/bar/foo`, then add a new test:
```ts
describe('glob config one level deep', () => {
    const tsConfigExcludeBar = {
      include: ['./**/*'],
      exclude: ['./src/bar.ts'],
    };
    const tsConfigIncludeAll = {
      include: ['./**/*'],
    };

    baseTests(tsConfigExcludeBar, tsConfigIncludeAll);
  });
```

Comment (armano2) [reply packages/typescript-estree/tests/lib/persistentParse.ts]: This is not correct, this issue happens only if extends is used
